--- libdbus-c++-0.9.0/include/dbus-c++/dispatcher.h.orig  2011-11-28 17:03:19.000000000 +0200 
+++ libdbus-c++-0.9.0/include/dbus-c++/dispatcher.h       2018-03-15 19:45:28.681636000 +0300 
@@ -266,7 +266,19 @@
   {
     return new Mx;
   }
-
+#ifndef DBUS_HAS_RECURSIVE_MUTEX 
+  static bool mutex_free(Mutex *mx) 
+  { 
+    delete mx; 
+    return true; 
+  } 
+ 
+  static bool mutex_lock(Mutex *mx) 
+  { 
+    mx->lock(); 
+    return true; 
+  } 
+#else 
   static void mutex_free(Mutex *mx)
   {
     delete mx;
@@ -276,7 +288,7 @@
   {
     mx->lock();
   }
-
+#endif
   static void mutex_unlock(Mutex *mx)
   {
     mx->unlock();
