--- acpica-unix-20170831.orig/generate/unix/iasl/Makefile	2017-08-31 18:57:55.000000000 +0300
+++ acpica-unix-20170831/generate/unix/iasl/Makefile	2017-09-20 13:24:56.401144030 +0300
@@ -338,24 +338,24 @@
 #
 $(OBJDIR)/aslcompilerlex.o :   $(OBJDIR)/aslcompilerlex.c
 	@echo "- " "Intermediate" $<
-	@$(CC) -c $(CFLAGS) -Wall -Werror -o$@ $<
+	@$(CC) -c $(CFLAGS) -Wall  -o$@ $<
 
 $(OBJDIR)/aslcompilerparse.o : $(OBJDIR)/aslcompilerparse.c
 	@echo "- " "Intermediate" $<
-	@$(CC) -c $(CFLAGS) -Wall -Werror -o$@ $<
+	@$(CC) -c $(CFLAGS) -Wall -o$@ $<
 
 $(OBJDIR)/dtparserlex.o :      $(OBJDIR)/dtparserlex.c
 	@echo "- " "Intermediate" $<
-	@$(CC) -c $(CFLAGS) -Wall -Werror -o$@ $<
+	@$(CC) -c $(CFLAGS) -Wall -o$@ $<
 
 $(OBJDIR)/dtparserparse.o :    $(OBJDIR)/dtparserparse.c
 	@echo "- " "Intermediate" $<
-	@$(CC) -c $(CFLAGS) -Wall -Werror -o$@ $<
+	@$(CC) -c $(CFLAGS) -Wall  -o$@ $<
 
 $(OBJDIR)/prparserlex.o :      $(OBJDIR)/prparserlex.c
 	@echo "- " "Intermediate" $<
-	@$(CC) -c $(CFLAGS) -Wall -Werror -o$@ $<
+	@$(CC) -c $(CFLAGS) -Wall  -o$@ $<
 
 $(OBJDIR)/prparserparse.o :    $(OBJDIR)/prparserparse.c
 	@echo "- " "Intermediate" $<
-	@$(CC) -c $(CFLAGS) -Wall -Werror -o$@ $<
+	@$(CC) -c $(CFLAGS) -Wall  -o$@ $<

