---
 multipathd/cli.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/multipathd/cli.c b/multipathd/cli.c
index d991cd0..b9930dc 100644
--- a/multipathd/cli.c
+++ b/multipathd/cli.c
@@ -109,7 +109,7 @@ set_handler_callback (uint64_t fp, int (*fn)(void *, char **, int *, void *))
 }
 
 int
-set_unlocked_handler_callback (unsigned long fp,int (*fn)(void *, char **, int *, void *))
+set_unlocked_handler_callback (uint64_t fp,int (*fn)(void *, char **, int *, void *))
 {
 	struct handler * h = find_handler(fp);
 
-- 
2.8.1